test/module/irohad/torii/torii_transport_command_test.cpp

```diff
@@ -0,0 +1,357 @@
+  EXPECT_CALL(*proto_tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Return(shared_model::validation::Answer{}));
+  EXPECT_CALL(*tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Return(error));
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(0);
+  EXPECT_CALL(*status_bus, publish(_)).Times(kTimes);
+
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service
+ *        and some number of valid transactions
+ *        and one stateless invalid tx
+ * @when calling ListTorii
+ * @then handleTransactionBatch is called kTimes - 1 times
+ *       and statelessInvalid status is published for invalid transaction
+ */
+TEST_F(CommandServiceTransportGrpcTest, ListToriiPartialInvalid) {
+  grpc::ServerContext context;
+  google::protobuf::Empty response;
+  const std::string kError = "some error";
+
+  iroha::protocol::TxList request{};
+  for (size_t i = 0; i < kTimes; ++i) {
+    request.add_transactions();
+  }
+
+  size_t counter = 0;
+  EXPECT_CALL(*proto_tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Return(shared_model::validation::Answer{}));
+  EXPECT_CALL(*tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Invoke([this, &counter, kError](const auto &) mutable {
+        shared_model::validation::Answer res;
+        if (counter++ == kTimes - 1) {
+          res.addReason(std::make_pair(kError, std::vector<std::string>{}));
+        }
+        return res;
+      }));
+  EXPECT_CALL(
+      *batch_factory,
+      createTransactionBatch(
+          A<const shared_model::interface::types::SharedTxsCollectionType &>()))
+      .Times(kTimes - 1);
+
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(kTimes - 1);
+  EXPECT_CALL(*status_bus, publish(_)).WillOnce(Invoke([&kError](auto status) {
+    EXPECT_THAT(status->statelessErrorOrCommandName(),
+                testing::HasSubstr(kError));
+  }));
+
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service and command_service with empty status stream
+ * @when calling StatusStream on transport
+ * @then Ok status is eventually returned without any fault
+ *       and nothing is written to the status stream
+ */
+TEST_F(CommandServiceTransportGrpcTest, StatusStreamEmpty) {
+  grpc::ServerContext context;
+  iroha::protocol::TxStatusRequest request;
+
+  EXPECT_CALL(*command_service, getStatusStream(_))
+      .WillOnce(Return(rxcpp::observable<>::empty<std::shared_ptr<
+                           shared_model::interface::TransactionResponse>>()));
+
+  ASSERT_TRUE(transport_grpc->StatusStream(&context, &request, nullptr).ok());
+}
+
+/**
+ * @given torii service with changed timeout, a transaction
+ *        and a status stream with one NotRecieved status
+ * @when calling StatusStream
+ * @then ServerWriter call Write method and waits for initial timeout
```

Comment:
There is no assertion on `waits for initial timeout` present in the test. A task to make scheduler a component parameter will be great, however this statement should be removed from test description for now.
```suggestion
 * @then ServerWriter calls Write method
```